server/src/utils/events.ts

```diff
@@ -21,8 +22,16 @@ export const setupEventHandlers = (moduleRef: ModuleRef) => {
   const items: Item<EmitEvent>[] = [];
 
   // discovery
-  for (const Service of services) {
-    const instance = moduleRef.get<any>(Service);
+  for (const classRef of [...services, ...repositories.map((r) => r.provide)]) {
+    let instance;
+    try {
+      instance = moduleRef.get<any>(classRef);
+    } catch (error: any) {
+      if (!error.message.includes('scoped provider')) {
+        throw error;
+      }
+      continue;
+    }
```

Comment:
Repositories should not be using these events. You should call the corresponding repository method directly from a service, which subscribes to an event.

Reply:
The intent is to have the DB introspection done at startup after migrations run and before the database repository gets any calls that need to use this data. This gives quick and easy synchronous access to the table metadata across different repositories.

I think handling this in `runMigrations` instead would be simpler.

Reply:
Both those situations don't require repositories being wired to events.

Reply:
Right, so when the database service calls the repo's `runMigrations` method, the repo can introspect the DB before returning without being wired to an event directly.

Reply:
Right, the repo doesn't need to _directly_ listen for it. You could do it in `runMigrations` or you could add a second interface method, like `init`, which could do it.